recipes/qt/5.x.x/conanfile.py

```diff
@@ -662,6 +666,43 @@ def package(self):
         for fl in glob.glob(os.path.join(self.package_folder, "lib", "*Qt5Bootstrap*")):
             os.remove(fl)
 
+        for m in os.listdir(os.path.join(self.package_folder, "lib", "cmake")):
+            module = os.path.join(self.package_folder, "lib", "cmake", m, "%sMacros.cmake" % m)
+            if not os.path.isfile(module):
+                tools.rmdir(os.path.join(self.package_folder, "lib", "cmake", m))
+
+
+        extension = ""
+        if self.settings.os == "Windows":
+            extension = ".exe"
+        filecontents = "set(QT_CMAKE_EXPORT_NAMESPACE Qt5)\n"
+        filecontents += "set(QT_VERSION_MAJOR %d)\n" % int(self.version.split(".")[0])
+        filecontents += "set(QT_VERSION_MINOR %d)\n" % int(self.version.split(".")[1])
+        filecontents += "set(QT_VERSION_PATCH %d)\n" % int(self.version.split(".")[2])
+        targets = {}
+        targets["Core"] = ["moc", "rcc", "qmake"]
+        targets["DBus"] = ["qdbuscpp2xml", "qdbusxml2cpp"]
+        if self.options.widgets:
+            targets["Widgets"] = ["uic"]
+        if self.options.qttools:
+            targets["Tools"] = ["qhelpgenerator", "qcollectiongenerator", "qdoc", "qtattributionsscanner"]
+            targets[""] = ["lconvert", "lrelease", "lupdate"]
+        if self.options.qtremoteobjects:
+            targets["RemoteObjects"] = ["repc"]
+        if self.options.qtscxml:
+            targets["Scxml"] = ["qscxmlc"]
+        for namespace, targets in targets.items():
+            for target in targets:
+                filecontents += textwrap.dedent("""\
+                    if(NOT TARGET ${{QT_CMAKE_EXPORT_NAMESPACE}}::{0})
+                        add_executable(${{QT_CMAKE_EXPORT_NAMESPACE}}::{0} IMPORTED)
+                        set_target_properties(${{QT_CMAKE_EXPORT_NAMESPACE}}::{0} PROPERTIES IMPORTED_LOCATION ${{CMAKE_CURRENT_LIST_DIR}}/../../../bin/{0}{1})
+                        set(Qt5{2}_{3}_EXECUTABLE ${{QT_CMAKE_EXPORT_NAMESPACE}}::{0})
+                    endif()
+                    """.format(target, extension, namespace, target.upper()))
```

Comment:
```suggestion
                    if(NOT TARGET ${{QT_CMAKE_EXPORT_NAMESPACE}}::{target})
                        add_executable(${{QT_CMAKE_EXPORT_NAMESPACE}}::{target} IMPORTED)
                        set_target_properties(${{QT_CMAKE_EXPORT_NAMESPACE}}::{target} PROPERTIES IMPORTED_LOCATION ${{CMAKE_CURRENT_LIST_DIR}}/../../../bin/{target}{ext})
                        set(Qt5{namespace}_{uppercase_target}_EXECUTABLE ${{QT_CMAKE_EXPORT_NAMESPACE}}::{target})
                    endif()
                    """.format(target=target, ext=extension, namespace=namespace, uppercase_target=target.upper()))
```

Less error prone I think